src/script/descriptor.cpp

```diff
@@ -696,14 +696,17 @@ NODISCARD bool ParseKeyPath(const std::vector<Span<const char>>& split, KeyPath&
             hardened = true;
         }
         uint32_t p;
-        if (!ParseUInt32(std::string(elem.begin(), elem.end()), &p) || p > 0x7FFFFFFFUL) return false;
+        if (!ParseUInt32(std::string(elem.begin(), elem.end()), &p) || p > 0x7FFFFFFFUL) {
+            error = strprintf("Key path value %u is out of range", p);
```

Comment:
Nit: "not valid" instead of "out of range"? Or split the check?

Reply:
I've split the check

Reply:
Now it's no longer split.

Reply:
> Now it's no longer split.

No??